### PR TITLE
fix: use subquery to improve performance

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We found that the new Session Replay filters were super slow

## Changes

Switching back from a join to subqueries in an attempt to improve performance

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Lots more tests pass

### CODE REVIEW

File: posthog/session_recordings/queries/session_recording_list_from_filters.py [hunk 1]
```diff
@@ -245,17 +228,121 @@ def _having_predicates(self) -> ast.And | Constant:
...
+                ast.CompareOperation(
+                    op=ast.CompareOperationOp.LtEq,
+                    left=ast.Field(chain=["timestamp"]),
+                    right=ast.Constant(value=self._filter.date_to + timedelta(hours=12)),
+                )
+            )
+
+        return ast.And(exprs=exprs)
+
+    def _where_predicates(self) -> ast.Expr:
```
Comment (daibhin) [inline posthog/session_recordings/queries/session_recording_list_from_filters.py:312]: I specifically left the [`persons_sub_query`](https://github.com/PostHog/posthog/blob/master/posthog/session_recordings/queries/session_recording_list_from_replay_summary.py#L548-L564) out of this where clause. It seemed related to PoE (based on the underlying [ActorsQuery](https://github.com/PostHog/posthog/blob/5fe209535efc9758b0414c12e8c9cafc3e0c9d56/posthog/session_recordings/queries/session_recording_list_from_replay_summary.py#L160-L260)).

I didn't have full context to know whether it was needed (maybe @pauldambra you know why it was added) or perhaps HogQL takes care of things under the hood

File: posthog/hogql/test/test_printer.py [hunk 2]
```diff
@@ -1212,7 +1212,7 @@ def test_print_global_settings(self):
         )
         self.assertEqual(
             printed,
-            f"SELECT 1 FROM events WHERE equals(events.team_id, {self.team.pk}) LIMIT 10000 SETTINGS readonly=2, max_execution_time=10, allow_experimental_object_type=1, format_csv_allow_double_quotes=0, max_ast_elements=1000000, max_expanded_ast_elements=1000000",
+            f"SELECT 1 FROM events WHERE equals(events.team_id, {self.team.pk}) LIMIT 10000 SETTINGS readonly=2, max_execution_time=10, allow_experimental_object_type=1, format_csv_allow_double_quotes=0, max_ast_elements=1000000, max_expanded_ast_elements=1000000, max_query_size=524288",
```
Comment (daibhin) [inline posthog/hogql/test/test_printer.py:1215]: I am not sure why I needed to add this but it was failing locally for me on `master` too. Probably worth a second opinion from someone on @PostHog/team-product-analytics

Comment (mariusandra) [reply posthog/hogql/test/test_printer.py:1215]: Apologies, we merged in some unsnapshotted changes. These should be reverted by now (merge with master)

**Comment by pauldambra** [comment]
https://github.com/PostHog/posthog/issues/22054

File: posthog/session_recordings/queries/session_recording_list_from_filters.py (hunk 1 above)
Comment (pauldambra) [reply posthog/session_recordings/queries/session_recording_list_from_filters.py:322]: _hopefully_ it's covered by HogQL

I know there were some customers that couldn't filter on person properties without PoE working but we can make sure we test with them before rolling out the flag and get some test cases here to prove the queries (if they don't exist already)

**Review by pauldambra — APPROVED** [review]
🚢

**Pull request merged**